src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultingString = '';
+
+    for (let i = 0; i < this.length; i += 1) {
+      let item = '';
+
+      if (this[i] !== null && this[i] !== undefined) {
+        item = this[i];
+      }
+      // const item = this[i] === null || this[i] === undefined ? "" : this[i];
+
+      if (i === this.length - 1) {
+        resultingString += `${item}`;
+        continue;
+      }
```

Comment:
can you merge these 2 conditions without creating additional `item` variable?
you can check the value is not `null` or `undefined` **and** not the last in array, then add `item` and `separator`